backend/internal/service/matching/execute.go

```diff
@@ -0,0 +1,110 @@
+package matching
+
+import "day-trading-app/backend/internal/service/models"
+
+var s serviceImpl
+
+func ExecuteOrders(stockTxCommitQueue []models.StockMatch) {
+	for _, tx := range stockTxCommitQueue {
+		if isParent(tx) {
+			// Update stock transaction as completed
+			s.db.UpdateStockOrderStatus(tx.Order.UserName, tx.Order.StockTxID, tx.Order.OrderStatus)
```

Comment:
that is not going to work. You are creating a new var for the serviceImpl so it wont have db initialized. It's an empty struct.